### PR TITLE
[vcpkg] Avoid computing triplet ABIs for editable packages

### PR DESCRIPTION
**Pull request by ras0219**
This PR delays computation of `abi_info.triplet_abi` until after we perform a check for conditions that would disable computation of a package abi. This can save several seconds in the port inner-dev loop.

The specific examples of this case are `--editable` and `--head`, which will both disable our ability to compute ABIs. Furthermore, if a package without an ABI is installed, all transitive consumers also cannot have their ABIs computed.

Before:
```
PS C:\src\vcpkg> vcpkg install zlib:x64-windows --editable
Computing installation plan...
The following packages will be built and installed:
    zlib[core]:x64-windows
Detecting compiler hash for triplet x64-windows...
Starting package 1/1: zlib:x64-windows
Building package zlib[core]:x64-windows...
-- Using cached C:/src/vcpkg/downloads/zlib1211.tar.gz
-- Using source at C:/src/vcpkg/buildtrees/zlib/src/1.2.11-63309e48e5
-- Configuring x64-windows
...
```
After:
```
PS C:\src\vcpkg> vcpkg install zlib:x64-windows --editable
Computing installation plan...
The following packages will be built and installed:
    zlib[core]:x64-windows
Starting package 1/1: zlib:x64-windows
Building package zlib[core]:x64-windows...
-- Using cached C:/src/vcpkg/downloads/zlib1211.tar.gz
-- Using source at C:/src/vcpkg/buildtrees/zlib/src/1.2.11-63309e48e5
-- Configuring x64-windows
...
```

+@qis who may be particularly happy about this change :)

### CODE REVIEW

File: toolsrc/src/vcpkg/build.cpp [hunk 1]
```diff
@@ -813,16 +811,49 @@ namespace vcpkg::Build
         }
     }
 
+    struct AbiTagAndFile
+    {
+        const std::string& triplet_abi;
```
Comment (BillyONeal) [inline toolsrc/src/vcpkg/build.cpp:816]: const-or-reference-members are scary due to

* CWG1116 “Aliasing of union members”, and
* CWG1776 “Replacement of class objects containing reference members”

could this be a pointer instead?

**Review by BillyONeal — APPROVED** [review]
Should this have a test?

**Comment by ras0219** [comment]
Adding a test would require mocking `System::cmd_execute()` and friends; I think that would be too large of an infrastructure change for now given the size of this PR.

**Comment by BillyONeal** [comment]
> Adding a test would require mocking `System::cmd_execute()` and friends;

I don't mean a unit test, I mean in the e2e tests.

**Comment by qis** [comment]
@ras0219 Works as advertised. Thank you!

**Pull request merged**